src/metabase/query_processor/middleware/desugar.clj

```diff
@@ -4,21 +4,54 @@
    [metabase.legacy-mbql.predicates :as mbql.preds]
    [metabase.legacy-mbql.schema :as mbql.s]
    [metabase.legacy-mbql.util :as mbql.u]
+   [metabase.lib.metadata :as lib.metadata]
    [metabase.lib.util.match :as lib.util.match]
+   [metabase.query-processor.store :as qp.store]
    [metabase.util.malli :as mu]))
 
+(defn- add-missing-field-base-types
```

Comment:
This function ensures that fields with id and missing base type have base in field options prior to dusugaring. That is necessary for correct `:is-empty` and `:not-empty` desugaring.